python-package/xgboost/core.py

```diff
@@ -217,6 +222,51 @@ def c_array(ctype, values):
     return (ctype * len(values))(*values)
 
 
+def _use_columnar_initializer(data):
+    '''Whether should we use columnar format initializer (pass data in as
+json string).  Currently cudf is the only valid option.'''
+    if CUDF_INSTALLED and isinstance(data, CUDF_DataFrame):
+        return True
+    return False
+
+
+def _extract_interface_from_cudf(df, is_info):
+    '''This function should be upstreamed to cudf.'''
+    if not _use_columnar_initializer(df):
+        raise ValueError('Only cudf is supported for initializing as json ' +
+                         'columnar format.  For other libraries please ' +
+                         'refer to specific API.')
+
+    def get_interface(obj):
+        return obj.mem.__cuda_array_interface__
+
+    array_interfaces = []
+    for col in df.columns:
+        data = df[col]._column._data  # pylint: disable=W0212
```

Comment:
You shouldn't need to reach into internal attributes, this will return you the same Buffer object via properties
```suggestion
        data = df[col].data
```

Reply:
@kkraus14 Thanks, good to know.